fibdrv.c

```diff
@@ -27,8 +28,9 @@ static int major = 0, minor = 0;
 
 static long long fib_sequence(long long k)
 {
-    /* FIXME: C99 variable-length array (VLA) is not allowed in Linux kernel. */
-    long long f[k + 2];
+    long long *f = kmalloc(sizeof(*f) * (k + 2), GFP_KERNEL);
+    if (!f)
+        return -1;
```

Comment:
The return value `-1` should be documented, otherwise it is nonsensical as for the math definition.

Reply:
I supposed `errno` would be appropriate to describe the situation?

`   return -ENOMEM;`